MovingBlocks/Terasology
# refactor: merge world config in setup screen

### Contains

Continues Advanced Game Setup (AGS) overhaul started in #5118 and #5122.
Merges `WorldSetupScreen` into `UniverseSetupScreen`.

Fixes #4810 as a side effect.

### How to test

1. Start Terasology
2. Start creating a world
3. Go to Advanced Game Setup (AGS)
4. Continue to Universe Setup
5. Play around with the screen, check for unclear UI elements, unexpected UI behavior, etc.
6. Verify that the game can be started

### Outstanding before merging

- [x] Trigger a preview update when configuration is changed
- [x] clean up code where necessary
- [x] improve documentation where necessary
